scripts/globals/weaponskills/energy_drain.lua

```diff
@@ -1,11 +1,35 @@
 -----------------------------------
 -- Energy Drain
 -----------------------------------
+require("scripts/globals/status")
+require("scripts/settings/main")
+require("scripts/globals/weaponskills")
+-----------------------------------
 local weaponskill_object = {}
 
 weaponskill_object.onUseWeaponSkill = function(player, target, wsID, tp, primary, action, taChar)
-   -- TODO: Should steal MP based on TP and not wake the mob involved.
-    return 1, 0, false, 65
+   -- TODO: Should not wake the mob involved. Not tested on sleeping mob
+
+    local damage = 60 -- Base mp absorbtion from http://wiki.ffo.jp/html/686.html saying it was between 60 and 120 and always higher than Energy Steal
+
+    -- params.ftp100 = 1 params.ftp200 = 1.5 params.ftp300 = 2 -- to follow earlier statement of been always higher than Energy Steal
+    -- params.mnd_wsc = 1
+
+    local damagemod = damage * ((tp - 1000) / 2000 + 1) -- Simplified formula for getting difference in tp return from 1000 tp to 3000 tp
+
+    if xi.settings.USE_ADOULIN_WEAPON_SKILL_CHANGES then
+        local attMind = player:getStat(xi.mod.MND)
+        damagemod = damagemod + attMind * 1
```

Comment:
Why are we multiplying by 1 here?

This 2 lines could be simplified to:
`damagemod = damagemod + player:getStat(xi.mod.MND)`

Reply:
I might have removed some of my explaination while cleaning up.
I couldn't find the code that would let me make this mp drain ws work from the weaponskills.lua and I don't know or where in the weaponskills.lua it should be added so I did all the relevent calculation from the ws file. This ws does no damage and only drain mp from mob with mp that aren't undead. The time 1 is really not useful it's a place older in the formula to implement the ws with the commented out parameter params.mnd_wsc = 1.

Reply:
there is no need to pass it to a variable at that point so I can remove it.

Reply:
Your point about resistance is a good question. While I'm not 100% sure I believe it is a straight mp steal that can not be resisted. The description on https://ffxiclopedia.fandom.com/wiki/Energy_Drain does specify that it's not align with elemental gorgets or elemental belts and does not have skill chain property. But it does have a darkness element.